src/test/java/com/google/sps/servlets/ContestServletTest.java

```diff
@@ -0,0 +1,178 @@
+
+  private Entity electionEntityOne;
+  private Entity contestEntityOne;
+  private Entity contestEntityTwo;
+
+  @Rule public MockitoRule mockitoRule = MockitoJUnit.rule();
+
+  @Before
+  public void setUp() {
+    helper.setUp();
+    electionEntityOne = new Entity("Election");
+    electionEntityOne.setProperty("id", "9999");
+    electionEntityOne.setProperty("name", "myElection");
+    electionEntityOne.setProperty("scope", "myScope");
+    electionEntityOne.setProperty("date", "myDate");
+    electionEntityOne.setProperty("contests", new HashSet<Long>());
+    electionEntityOne.setProperty("propositions", new HashSet<Long>());
+
+    contestEntityOne = new Entity("Contest");
+    contestEntityOne.setProperty("name", "myFirstContest");
+    contestEntityOne.setProperty("candidates", CANDIDATE_ID_SET);
+    contestEntityOne.setProperty("description", "This contest is important.");
+
+    contestEntityTwo = new Entity("Contest");
+    contestEntityTwo.setProperty("name", "mySecondContest");
+    contestEntityTwo.setProperty("candidates", new HashSet<Long>());
+    contestEntityTwo.setProperty("description", "This contest is also important.");
+  }
+
+  @Test
+  public void singleElection_singleContest_testDoGet() throws Exception {
+    Entity electionEntity = new Entity("Election");
+    Entity contestEntity = new Entity("Contest");
+
+    electionEntity.setPropertiesFrom(electionEntityOne);
+    contestEntity.setPropertiesFrom(contestEntityOne);
+
+    DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+    long contestId = ds.put(contestEntity).getId();
+
+    Collection<Long> contestSet = (Collection<Long>) electionEntity.getProperty("contests");
+    contestSet.add(contestId);
+    ds.put(electionEntity);
+
+    when(httpServletRequest.getParameter("electionId")).thenReturn("9999");
+    when(httpServletResponse.getWriter()).thenReturn(printWriter);
+
+    ContestsServlet contestServlet = new ContestsServlet();
+    contestServlet.doGet(httpServletRequest, httpServletResponse);
+
+    verify(printWriter)
+        .println(
+            "[{\"name\":\"myFirstContest\",\"candidates\":[1,2,3],\"description\":\"This contest is important.\"}]");
+  }
+
+  @Test
+  public void singleElection_twoContest_testDoGet() throws Exception {
+    Entity electionEntity = new Entity("Election");
+    Entity firstContestEntity = new Entity("Contest", 1);
+    Entity secondContestEntity = new Entity("Contest", 2);
+
+    electionEntity.setPropertiesFrom(electionEntityOne);
+    firstContestEntity.setPropertiesFrom(contestEntityOne);
+    secondContestEntity.setPropertiesFrom(contestEntityTwo);
+
+    DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+    long contestOneId = ds.put(firstContestEntity).getId();
+    long contestTwoId = ds.put(secondContestEntity).getId();
+
+    HashSet<Long> contestSet = (HashSet<Long>) electionEntityOne.getProperty("contests");
```

Comment:
I thought we couldn't cast into HashSet? maybe you can use a HashSet constructor instead?

Feel free to just resolve if this runs successfully :)